### PR TITLE
move tree enabled for pages

### PR DESCRIPTION
**Pull request by paddeo**

### CODE REVIEW

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 1]
```diff
@@ -19,183 +19,266 @@
...
+        }
+        return SUCCESS;
+    }
+
+    protected String checkMovePage(String selectedNode, String parentPageCode) {
+        IPage currentPage = this.getPage(selectedNode);
+        if (null == currentPage) {
+            _logger.info("Required a selected node");
+            this.addActionError(this.getText("error.page.selectPage"));
+            return "pageTree";
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: why magic words?

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 2]
```diff
@@ -19,183 +19,266 @@
...
+        }
+        if (currentPage.getCode().equals(this.getPageManager().getRoot().getCode())) {
+            _logger.info("Root page cannot be moved");
+            this.addActionError(this.getText("error.page.move.rootNotAllowed"));
+            return "pageTree";
+        }
+        if ("".equals(parentPageCode) || null == this.getPageManager().getPage(parentPageCode)) {
+            this.addActionError(this.getText("error.page.move.selectPageParent"));
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: missing logger

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 3]
```diff
@@ -19,183 +19,266 @@
...
+        if (parent.getCode().equals(currentPage.getParentCode())) {
+            _logger.debug("trying to move a node under it's own parent..");
+            this.addActionError(this.getText("error.page.move.parentUnderChild.notAllowed"));
+            return "pageTree";
+        }
+        if (parent.isChildOf(selectedNode)) {
+            List<String> args = new ArrayList<String>();
+            args.add(parent.getCode());
+            args.add(selectedNode);
+            this.addActionError(this.getText("error.page.move.parentUnderChild.notAllowed"));
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: missing logger?

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 4]
```diff
@@ -19,183 +19,266 @@
...
+            return "pageTree";
+        }
+        //group check
+        if (!currentPage.getGroup().equals(Group.FREE_GROUP_NAME)
+                && !currentPage.getGroup().equals(parent.getGroup())) {
+            this.addActionError(this.getText("error.page.move.differentRestriction.notAllowed"));
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: missing logger

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 5]
```diff
@@ -19,183 +19,266 @@
...
+        }
+        IPage parent = this.getPage(parentPageCode);
+        if (null == parent) {
+            _logger.info("Required a selected node");
+            this.addActionError(this.getText("error.page.selectPageParent"));
+            return "pageTree";
+        }
+        if (parent.getCode().equals(currentPage.getParentCode())) {
+            _logger.debug("trying to move a node under it's own parent..");
+            this.addActionError(this.getText("error.page.move.parentUnderChild.notAllowed"));
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: these two lines of code keep repeating themselves over and over: should this be its own private method?

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 6]
```diff
@@ -19,183 +19,266 @@
...
+            return "pageTree";
+        }
+        //group check
+        if (!currentPage.getGroup().equals(Group.FREE_GROUP_NAME)
+                && !currentPage.getGroup().equals(parent.getGroup())) {
+            this.addActionError(this.getText("error.page.move.differentRestriction.notAllowed"));
+            return "pageTree";
+        }
+        return null;
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: this is a brand new method that is either returning pageTree if an error occurs or null if it is successful.
Couldn't this just be returning a boolean at this point and have the method consuming this return pageTree?

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 7]
```diff
@@ -19,183 +19,266 @@
...
+        IPage selectedPage = this.getPageManager().getDraftPage(selectedNode);
+        if (null == selectedPage) {
+            this.addActionError(this.getText("error.page.selectedPage.null"));
+            return "pageTree";
+        }
+        if (!this.isUserAllowed(selectedPage)) {
+            this.addActionError(this.getText("error.page.userNotAllowed"));
+            return "pageTree";
+        }
+        return null;
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: why there is no logging on this method? btw the body of each if is the same one used in the method above: shouldn't we definitely move the body to its own private? DRY

File: admin-console/src/test/java/com/agiletec/apsadmin/portal/TestPageTreeAction.java [hunk 8]
```diff
@@ -30,296 +29,338 @@
...
+        Collection<String> errors = this.getAction().getActionErrors();
+        assertEquals(1, errors.size());
+
+        pageToMoveCode = "errorpage";
+        this.initAction("/do/Page", "moveDown");
+        this.addParameter("selectedNode", pageToMoveCode);
+        result = this.executeAction();
+        assertEquals(Action.SUCCESS, result);
+        errors = this.getAction().getActionErrors();
+        assertEquals(1, errors.size());
```
Comment (nicpuddu) [inline admin-console/src/test/java/com/agiletec/apsadmin/portal/TestPageTreeAction.java]: should we not also be checking which error has been returned? this applies to all the tests in this file

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/page.xml [hunk 9]
```diff
@@ -5,483 +5,514 @@
...
-				<param name="root">lastUpdatePagesResponse</param>
-			</result>
-			<result name="failure" type="json">
-				<param name="root">lastUpdatePagesResponse</param>
-			</result>
-			<interceptor-ref name="jsonStack" />
-			<interceptor-ref name="entandoDefaultStack">
-				<param name="requestAuth.requiredPermission">managePages</param>
-			</interceptor-ref>
-		</action>
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/page.xml]: are these two actions really supposed to be deleted?

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java (hunk 1 above)
Comment (paddeo) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: entando standard, I suppose.

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java (hunk 2 above)
Comment (paddeo) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: fixed

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java (hunk 3 above)
Comment (paddeo) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: fixed.

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java (hunk 4 above)
Comment (paddeo) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: fixed.

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java [hunk 10]
```diff
@@ -19,183 +19,277 @@
...
+            _logger.error("error in movePage", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    public String moveTree() {
+        String selectedNode = this.getSelectedNode();
+        String parentPageCode = this.getRequest().getParameter("parentPageCode");
+        if (null == parentPageCode) {
```
Comment (stefanopuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: should be safer with StringUtils.isNotBlank

Comment (paddeo) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java]: fixed.

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/page.xml [hunk 11]
```diff
@@ -5,483 +5,545 @@
...
+            </result>
+            <result name="input" type="json">
+                <param name="root">pagesStatusResponse</param>
+            </result>
+            <result name="failure" type="json">
+                <param name="root">pagesStatusResponse</param>
+            </result>
+            <interceptor-ref name="jsonStack" />
+            <interceptor-ref name="entandoDefaultStack">
+                <param name="requestAuth.requiredPermission">managePages</param>
```
Comment (stefanopuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/page.xml]: should be enterBackend

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/page.xml [hunk 12]
```diff
@@ -5,483 +5,545 @@
...
+            </result>
+            <result name="input" type="json">
+                <param name="root">lastUpdatePagesResponse</param>
+            </result>
+            <result name="failure" type="json">
+                <param name="root">lastUpdatePagesResponse</param>
+            </result>
+            <interceptor-ref name="jsonStack" />
+            <interceptor-ref name="entandoDefaultStack">
+                <param name="requestAuth.requiredPermission">managePages</param>
```
Comment (stefanopuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/page.xml]: should be enterBackend

**Comment by nicpuddu** [comment]
:shipit:

**Pull request merged**